kura/org.eclipse.kura.driver.opcua.provider/src/main/java/org/eclipse/kura/internal/driver/opcua/OpcUaDriver.java

```diff
@@ -106,79 +102,24 @@
-            .newConcurrentHashMap();
-
-    // initialize Milo Stack class loader
-    static {
-        try {
-            final URL url = new URL("platform:/plugin/org.eclipse.kura.driver.opcua.provider/lib/stack-core-0.1.0.jar");
-            final Optional<URLConverter> converter = getURLConverter(url);
-            Optional<URL> fileUrl = Optional.empty();
-            if (converter.isPresent()) {
-                fileUrl = Optional.of(converter.get().toFileURL(url));
-            }
-            Optional<ClassLoader> classLoader = Optional.empty();
-            if (fileUrl.isPresent()) {
-                classLoader = Optional
-                        .of(new URLClassLoader(new URL[] { fileUrl.get() }, OpcUaDriver.class.getClassLoader()));
-            }
-            if (classLoader.isPresent()) {
-                Stack.setCustomClassLoader(classLoader.get());
-            }
-        } catch (final IOException e) {
-            logger.error(e.getMessage(), e);
-        }
-    }
-
     /** OPC-UA Client Connector */
     private OpcUaClient client;
 
     /** Dependency for password decryption. */
     private volatile CryptoService cryptoService;
 
-    /** flag to check if the driver is connected. */
-    private volatile boolean isConnected;
-
     /** OPC-UA Configuration Options. */
     private OpcUaOptions options;
 
-    /**
-     * Remove the already tracked {@link URLConverter} service trackers
-     */
-    private static void closeURLConverterTrackers() {
-        if (!urlConverterTrackers.isEmpty()) {
-            for (final ServiceTracker<Object, URLConverter> tracker : urlConverterTrackers.values()) {
-                tracker.close();
-            }
-            urlConverterTrackers = CollectionUtil.newConcurrentHashMap();
-        }
-    }
+    private AtomicBoolean isBusy = new AtomicBoolean();
 
-    /**
-     * Returns the {@link URLConverter} service for the given URL.
-     */
-    public static Optional<URLConverter> getURLConverter(final URL url) {
-        final BundleContext ctx = FrameworkUtil.getBundle(OpcUaDriver.class).getBundleContext();
-        if ((url == null) || (ctx == null)) {
-            return Optional.empty();
-        }
-        final String protocol = url.getProtocol();
-        ServiceTracker<Object, URLConverter> tracker = urlConverterTrackers.get(protocol);
-        if (isNull(tracker)) {
-            final String urlFilter = "(&(objectClass=" + URLConverter.class.getName() + ")(protocol=" + protocol + "))";
-            Filter filter = null;
-            try {
-                filter = ctx.createFilter(urlFilter);
-            } catch (final InvalidSyntaxException e) {
-                return Optional.empty();
-            }
-            tracker = new ServiceTracker<>(ctx, filter, null);
-            tracker.open();
-            urlConverterTrackers.put(protocol, tracker);
+    private <U> U runSafe(Future<U> future) throws ExecutionException, InterruptedException, TimeoutException {
```

Comment:
Another argument could be that adding preconditions in private methods adds unnecessary overhead: you have an additional method invocation and null check for every requireNonNull(), this can be undesirable in particular for short and frequently called methods.
If the precondition is not satisfied you will end up with a NullPointerException, but the same happens if you do not add the precondition and you try to use a null pointer, the second case is faster however because you don't have unnecessary checks.